code/datums/components/proximity.dm

```diff
@@ -0,0 +1,56 @@
+TYPEINFO(/datum/component/proximity)
+	initialization_args = list(
+		ARG_INFO("enabled", DATA_INPUT_BOOL, "State of detection.", TRUE),
+		ARG_INFO("enabled", DATA_INPUT_NUM, "Tiles out to detect.", 1)
+	)
+/datum/component/proximity
+	var/enabled = TRUE
+	var/range = 1
+	VAR_PRIVATE/list/turf/listening_to //this feels really unclean but the other way involves a bunch of signals and it makes me :( -cringe
+
+/datum/component/proximity/Initialize(enabled = TRUE, range = 1)
+	..()
+	if(!isatom(src.parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.enabled = enabled
+	src.range = range
+	src.listening_to = list()
+
+/datum/component/proximity/RegisterWithParent()
+	var/atom/A = src.parent
+	var/turf/center = get_turf(A)
+	if(ismovable(src.parent))
+		RegisterSignal(src.parent, XSIG_MOVABLE_TURF_CHANGED, PROC_REF(parent_moved))
```

Comment:
Using this complex signal allows proximity stuff to function contained - previous hasproximity stuff would only track turfs when directly on a turf.
Should either revise, or make it explicit in documentation/PR that this is not equivalent behaviour

Reply:
ill note this as not equivalent.

Reply:
or could maybe add a way to have equivalent behavior hm.